firebase-firestore/src/test/java/com/google/firebase/firestore/local/LocalStoreTestCase.java

```diff
@@ -849,17 +873,28 @@ public void testCanExecuteMixedCollectionQueries() {
     allocateQuery(query);
     assertTargetId(2);
 
+    localStore.executeQuery(query);
```

Comment:
This isn't awful but please consider making a separate test that specifically exercises interesting cases rather than making additional assertions as a side-effect of some other scenario. In the current configuration it becomes difficult to understand what part of the test is essential to verifying mixed collection queries vs what's essential to verifying the stats.

Reply:
I basically wanted to write the exact same test that we have here. I did remove the checks in 877 and 888, which should improve the readability of this test.

I can create a separate test but it would likely contain most lines from this test. What do you think?

Reply:
The worry I essentially have is that we end up wanting to change this test to verify some additional case and then get tripped up around the interaction with the counts and then it's not clear if changing the counts is the right thing to do.

I couldn't find a reference to it quickly, but some hardline testing guru types advocate for only having one assertion per test case essentially on these grounds.

Most of these tests are copies of each other in some way so I don't think that's much of an issue.

Reply:
Updated the PR to add a separate test.